src/plugins/layout/shadowdom.ts

```diff
@@ -88,7 +92,11 @@ export class ShadowDom {
 
   public updateShadowNode(index: number, newLayout?: ILayoutState) {
     let shadowNode = this.getShadowNode(index);
-    assert(!!shadowNode, "updateShadowNode");
+
+    assert(!!shadowNode, "updateShadowNode", "shadowNode is missing");
```

Comment:
Add more useful message to indicate what is missing. Add a way to show the element and its location in the tree and include that in the message.